### PR TITLE
ci: update tests workflow to use centralised reusable workflow and auto-suggest format changes

### PR DESCRIPTION
**Pull request by thazhemadam**
Follow up to #153.

### CODE REVIEW

**Comment by avik-pal** [comment]
Are the failures stemming from `__solve(::NonlinearProblem, ::<SimpleNonlinearSolveAlg>)` being removed in favor of `ImmutableNonlinearProblem`?

**Comment by ChrisRackauckas** [comment]
The current failure is just the depwarn https://github.com/JuliaGPU/GPUCompiler.jl/issues/616 ?

**Comment by ChrisRackauckas** [comment]
The depwarns are from an old version, @avik-pal do you know why the old version is there?

**Comment by avik-pal** [comment]
https://github.com/EnzymeAD/Enzyme.jl/blob/e1edaf7482dee3be5bcf35afb7527cea808a1f07/Project.toml#L39 doesn't support the latest GPUCompiler yet. I think LLVM.jl 8 was the blocking issue there https://github.com/maleadt/LLVM.jl/issues/435 (cc @wsmoses)

**Comment by avik-pal** [comment]
(NOTE TO SELF) LuxTestUtils (https://github.com/SciML/DeepEquilibriumNetworks.jl/blob/de2c730e689e1de445c63e203e545a5725e58d04/Project.toml#L47) needs to be updated to the latest version

**Comment by ChrisRackauckas** [comment]
@wsmoses I thought you said v0.12.33 should do it? https://github.com/SciML/DeepEquilibriumNetworks.jl/actions/runs/10562146339/job/29259477769?pr=159#step:6:71

**Comment by wsmoses** [comment]
I said that bumped the gpucompiler compat, the deprecation from llvm.jl can’t be fixed til we drop old Julia’s with the new LTS (since the function is indicative of Julia version config for older Julia)

**Comment by ChrisRackauckas** [comment]
@avik-pal you got this as part of your other update?

**Comment by avik-pal** [comment]
yeah, I will merge that and then figure out the failure here.

**Comment by avik-pal** [comment]
Once `https://github.com/SciML/DeepEquilibriumNetworks.jl/actions/runs/10675542556/job/29587688915?pr=159#step:6:950` is fixed the tests will pass. Coming from Enzyme + GPUCompiler which will get fixed as mentioned in https://github.com/SciML/DeepEquilibriumNetworks.jl/pull/159#issuecomment-2312892359

File: .github/workflows/Tests.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref_name != github.event.repository.default_branch || github.ref != 'refs/tags/v*' }}
+
+jobs:
+  tests:
+    name: "Tests"
+    uses: "SciML/.github/.github/workflows/tests.yml@v1"
+    with:
+      group: "CPU"
```
Comment (avik-pal) [inline .github/workflows/Tests.yml:24]:
```suggestion
      BACKEND_GROUP: "CPU"
```

File: .github/workflows/Tests.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref_name != github.event.repository.default_branch || github.ref != 'refs/tags/v*' }}
+
+jobs:
+  tests:
+    name: "Tests"
+    uses: "SciML/.github/.github/workflows/tests.yml@v1"
+    with:
+      BACKEND_GROUP: "CPU"
```
Comment (avik-pal) [inline .github/workflows/Tests.yml]:
```suggestion
      group: "CPU"
```

**Pull request merged**